### PR TITLE
Road Create/Move/Delete routes

### PR DESCRIPTION
**Pull request by jameslaneconkling**
closes https://github.com/orma/openroads-vn-analytics/issues/253

Delete handler updates aren't transactional.

### CODE REVIEW

**Comment by jameslaneconkling** [comment]
weird, docs only show docs for 2 of the 3 routes (not `/properties/roads/:id/move`)

File: routes/road-properties.js [hunk 1]
```diff
@@ -18,29 +18,77 @@ function getHandler (req, res) {
...
+    if (err.constraint === 'road_properties_pkey') {
+      return res(Boom.conflict());
+    }
+
+    return res(Boom.badImplementation());
+  });
+}
+
+function moveHandler(req, res) {
+  knex('road_properties')
```
Comment (mileswwatkins) [inline routes/road-properties.js]: Won't this fail if `road_properties.id` is used in `point_properties.road_id` or `field_data_geometries.road_id`? I think you have to do something like:

- set related pp.road_id and fdg.road_id to null (save the `id`s of those records)
- update id in `road_properties`
- update related pp.road_id and fdg.road_id to the new `road_properties.road_id`

File: routes/road-properties.js [hunk 2]
```diff
@@ -18,29 +18,77 @@ function getHandler (req, res) {
...
+    .where({ id: req.params.road_id })
+    .update({
+      id: req.payload.id
+    })
+  .then(function(response) {
```
Comment (mileswwatkins) [inline routes/road-properties.js]: Also for moving, the `current_way_tags` values should be updated as well, if any exist. These aren't explicit foreign key relationships, but the OSM schema's `or_vpromms` key value is used as a join key (with `road_properties.id`) for several purposes.

So, for any `current_way.id`, if there is a `current_way_tags.k` of `or_vpromms`, and that row's `current_way_tags.v` is equal to the `road_properties.id` that's being moved, then update the `current_way_tags.v` value.

(Let's not worry about handling `current_way_tags.v` alteration for the `road_properties.id` delete handler.)

**Comment by jameslaneconkling** [comment]
@mileswwatkins  added `FOREIGN KEY ON UPDATE CASCADE` constraint on `point_properties` and `current_way_tags` tables, so tables stay consistent across `POST /properties/roads/{road_id}/move` calls.

Also added call to update `or_vpromms` keys in `current_way_tags` on `POST /properties/roads/{road_id}/move`

It wouldn't be hard to delete `or_vpromms` keys on `DELETE /properties/roads/{road_id}`.  Is this desired behavior?

**Comment by jameslaneconkling** [comment]
@mileswwatkins circleci failed w/ a cryptic error
```
Allocating a remote Docker Engine
Assigned Docker Engine request id: 4019530
  waiting .........................................

Got error while creating host: VM creation failed
We had an unexpected error preparing a VM for this build, potentially due to our infrastructure or cloud provider.  Please retry the build in a few minutes.
```

I don't have permission to rebuild.  Can you try running it agin (or give me write permission), and see how goes?

**Comment by mileswwatkins** [comment]
Logic looks good; going to try running locally now.

**Comment by mileswwatkins** [comment]
The road ID validation in the move handler validates `req.params.road_id` instead of `req.payload.id`. Everything else seems to be working properly!

**Pull request merged**